rnd/market/market/app.py

```diff
@@ -1,60 +1,80 @@
-
-from contextlib import asynccontextmanager
+import contextlib
 import os
-from dotenv import load_dotenv
-from fastapi import FastAPI
-from fastapi.middleware.gzip import GZipMiddleware
-from prisma import Prisma
+
+import dotenv
+import fastapi
+import fastapi.middleware.cors
+import fastapi.middleware.gzip
+import prisma
+import prometheus_fastapi_instrumentator
 import sentry_sdk
-from sentry_sdk.integrations.asyncio import AsyncioIntegration
-from sentry_sdk.integrations.fastapi import FastApiIntegration
-from sentry_sdk.integrations.starlette import StarletteIntegration
+import sentry_sdk.integrations.asyncio
+import sentry_sdk.integrations.fastapi
+import sentry_sdk.integrations.starlette
 
-from market.routes import agents
+import market.routes.admin
+import market.routes.agents
+import market.routes.search
 
-load_dotenv()
+dotenv.load_dotenv()
 
 if os.environ.get("SENTRY_DSN"):
     sentry_sdk.init(
-        dsn=os.environ.get("SENTRY_DSN"), 
-        # Set traces_sample_rate to 1.0 to capture 100%
-        # of transactions for performance monitoring.
+        dsn=os.environ.get("SENTRY_DSN"),
         traces_sample_rate=1.0,
-        # Set profiles_sample_rate to 1.0 to profile 100%
-        # of sampled transactions.
-        # We recommend adjusting this value in production.
         profiles_sample_rate=1.0,
         enable_tracing=True,
         environment=os.environ.get("RUN_ENV", default="CLOUD").lower(),
         integrations=[
-            StarletteIntegration(transaction_style="url"),
-            FastApiIntegration(transaction_style="url"),
-            AsyncioIntegration(),
+            sentry_sdk.integrations.starlette.StarletteIntegration(
+                transaction_style="url"
+            ),
+            sentry_sdk.integrations.fastapi.FastApiIntegration(transaction_style="url"),
+            sentry_sdk.integrations.asyncio.AsyncioIntegration(),
         ],
     )
 
-db_client = Prisma(auto_register=True)
+db_client = prisma.Prisma(auto_register=True)
 
 
-@asynccontextmanager
-async def lifespan(app: FastAPI):
+@contextlib.asynccontextmanager
+async def lifespan(app: fastapi.FastAPI):
     await db_client.connect()
     yield
     await db_client.disconnect()
 
 
-app = FastAPI(
-    title="Marketplace  API",
-    description=(
-        "AutoGPT Marketplace API is a service that allows users to share AI agents."
-    ),
+app = fastapi.FastAPI(
+    title="Marketplace API",
+    description="AutoGPT Marketplace API is a service that allows users to share AI agents.",
     summary="Maketplace API",
     version="0.1",
     lifespan=lifespan,
+    root_path="/api/v1/market",
+)
+
+app.add_middleware(fastapi.middleware.gzip.GZipMiddleware, minimum_size=1000)
+app.add_middleware(
+    middleware_class=fastapi.middleware.cors.CORSMiddleware,
+    allow_origins=[
+        # Currently, we allow only next.js dev server
+        "http://localhost:3000",
+        "http://127.0.0.1:3000",
```

Comment:
This will probably break the Dockerfile btw 
@aarushik93 

Not sure what CORS rules we would want with docker...

Reply:
If you create a ticket for me - I can take a look